### PR TITLE
check values in location updates for NaN before using them

### PR DESCRIPTION
**Pull request by sethalves**
- guard against NaN getting into the transform for entities and avatars

### CODE REVIEW

**Comment by sethalves** [comment]
https://app.asana.com/0/26485730907942/82901672699164

File: libraries/shared/src/AACube.cpp [hunk 1]
```diff
@@ -478,3 +478,13 @@ AACube& AACube::operator += (const glm::vec3& point) {
 
     return (*this);
 }
+
+bool AACube::containsNaN() const {
+    if ( _corner.x != _corner.x ||  _corner.y != _corner.y ||  _corner.z != _corner.z) {
```
Comment (ZappoMan) [inline libraries/shared/src/AACube.cpp]: I thought we had a portable isNaN() can you double check and use that one (yes, I believe it just does this same check, but still isNaN() is more clear.

File: libraries/shared/src/SpatiallyNestable.cpp [hunk 2]
```diff
@@ -314,6 +314,11 @@ glm::vec3 SpatiallyNestable::getPosition(int jointIndex, bool& success) const {
 }
 
 void SpatiallyNestable::setPosition(const glm::vec3& position, bool& success) {
+    // guard against introducing NaN into the transform
+    if (position.x != position.x || position.y != position.y || position.z != position.z) {
```
Comment (ZappoMan) [inline libraries/shared/src/SpatiallyNestable.cpp]: ibid

File: libraries/shared/src/SpatiallyNestable.cpp [hunk 3]
```diff
@@ -358,6 +363,15 @@ glm::quat SpatiallyNestable::getOrientation(int jointIndex, bool& success) const
 }
 
 void SpatiallyNestable::setOrientation(const glm::quat& orientation, bool& success) {
+    // guard against introducing NaN into the transform
+    if (orientation.x != orientation.x ||
```
Comment (ZappoMan) [inline libraries/shared/src/SpatiallyNestable.cpp]: ibid

**Comment by hifi-gustavo** [comment]
Build console output for commit: 451464cb474613b51ebe199e2533798e72aa1a79 available at: https://gist.github.com/2f69ea908558f0340dab<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6902.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6902.dmg

**Comment by sethalves** [comment]
I don't know what ibid means.

File: libraries/shared/src/SpatiallyNestable.cpp [hunk 4]
```diff
@@ -314,6 +314,11 @@ glm::vec3 SpatiallyNestable::getPosition(int jointIndex, bool& success) const {
 }
 
 void SpatiallyNestable::setPosition(const glm::vec3& position, bool& success) {
+    // guard against introducing NaN into the transform
+    if (isNaN(position.x) || isNaN(position.y) || isNaN(position.z)) {
```
Comment (ZappoMan) [inline libraries/shared/src/SpatiallyNestable.cpp]: doesn't glm have an isnan?

```
template<typename genType >
genType::bool_type 	isnan (genType const &x)
```

Can't you just do `if (glm::isnan(position)) { ...`

File: libraries/shared/src/SpatiallyNestable.cpp [hunk 5]
```diff
@@ -358,6 +363,12 @@ glm::quat SpatiallyNestable::getOrientation(int jointIndex, bool& success) const
 }
 
 void SpatiallyNestable::setOrientation(const glm::quat& orientation, bool& success) {
+    // guard against introducing NaN into the transform
+    if (isNaN(orientation.x) || isNaN(orientation.y) || isNaN(orientation.z) || isNaN(orientation.w)) {
```
Comment (ZappoMan) [inline libraries/shared/src/SpatiallyNestable.cpp]: ibid `glm::isnan()`

Comment (sethalves) [inline libraries/shared/src/SpatiallyNestable.cpp]: glm::isnan doesn't seem to take vec3 or quat.  I've added versions for these to glmhelpers.

**Comment by hifi-gustavo** [comment]
Build console output for commit: a9cf836cb047c4ed03b7457b5597812fb457e38c available at: https://gist.github.com/2f69ea908558f0340dab<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6902.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6902.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 673182c0918421e7e59ceb1a3e2e6a7769ea6daa available at: https://gist.github.com/2f69ea908558f0340dab<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6902.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6902.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2633cb1f272d4274929dae9cceae0b3731a2f70a available at: https://gist.github.com/2f69ea908558f0340dab<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6902.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6902.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2633cb1f272d4274929dae9cceae0b3731a2f70a available at: https://gist.github.com/2f69ea908558f0340dab<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6902.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6902.dmg

**Pull request merged**